File: codes/apna college/c++/primeFunc.cpp
#include<iostream>
#include<cmath>

using namespace std;

bool isPrime(int n)
{
    bool isPrime=true;
    if(n==1)
    {
        return false;
    }
    else if(n==2)
    {
        return true;
    }
    else{
        for(int i=2;i<=sqrt(n);i++)
        {
            if(n%i==0)
            {
                return false;
            }
        }
        return true;
    }
    // else{
    //     for(int i=2;i*i<=n;i++)//same as i to sqrt(n) --> i^2 to n
    //     {
    //         if(n%i==0)
    //         {
    //             return false;
    //         }
    //     }
    //     return true;
    // }

}

int main()
{
    int num;
    cout<<"Enter the number to check for number"<<endl;
    cin>>num;
    if(isPrime(num)==true)
    {
        cout<<num<<" is a prime number";
    }
    else{
        cout<<num<<" is not a prime number";
    }
}

File: codes/apna college/java/primeFunc.java
import java.util.Scanner;

public class primeFunc {

    public static void checkPrime(int n) {
        boolean isPrime = true;
        //corner cases
        if (n == 2) {
            System.out.println(n + " is a Prime Number");
        } else {
            for (int i = 2; i <= n-1; i++) {
                if (n % i == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.println(n + " is a Prime Number");
            } else {
                System.out.println(n + " is a Composite Number");
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;
        System.out.println("Enter the number to check for prime");
        n = sc.nextInt();
        checkPrime(n);
    }
}
